src/thirdparty/CMakeLists.txt

```diff
@@ -37,10 +37,21 @@ endif()
 
 set(CLANG_TIDY_EXE_TEMP ${CLANG_TIDY_EXE})
 set(CLANG_TIDY_EXE "")
+
 add_subdirectory(LSD)
 add_subdirectory(PoissonRecon)
 if(GPU_ENABLED)
     add_subdirectory(SiftGPU)
 endif()
 add_subdirectory(VLFeat)
+
+include(FetchContent)
+FetchContent_Declare(PoseLib
+  GIT_REPOSITORY    https://github.com/PoseLib/PoseLib.git
+  GIT_TAG           c23af0c51924e3337f7bb193bb72899ec6dd4d3d
+)
+message(STATUS "Configuring PoseLib...")
+FetchContent_MakeAvailable(PoseLib)
+message(STATUS "Configuring PoseLib... done")
```

Comment:
@pablospe Here. IMHO the alternative (PoseLib as submodule) is less convenient because it requires users to install an additional dependency or to add `--recursive` when cloning colmap (which is easy to forget).

Reply:
yes, you are right and I agree. I was actually suggesting `FetchContent`.

Reply:
Well... actually thinking it could be an option and see the opinions.

Reply:
Viktor's PR has been merged into master, see https://github.com/PoseLib/PoseLib/pull/86

Reply:
Please, consider a scenario where someone is working on modifications into PoseLib which are used/assessed in context of colmap and such work is performed on local machine.

If i understand correctly, in case of `FetchContent` (if we ignore relatively new `Dependency Providers` functionality) it would be necessary to modify a call to `FetchContent_Declare` in order to point to a local folder with modified PoseLib sources, also, for those who use IDE, working on such sources would require one more IDE context.

In case of PoseLib dependency satisfied via git submodule no "temporary" modifications to colmap's cmake scripts would be required and such work could be performed in single IDE context.

Reply:
I am curious about what you meant exactly by ignoring `Dependency Providers`. Maybe you can expand more on this, I was reading the doc and it seems that it is not ignored (but I am not sure what you refer):
https://cmake.org/cmake/help/latest/command/cmake_language.html#dependency-providers

My concern with `FetchContent` is how it plays with code autocomplete (like in vscode or another IDE), since you need to configure it first with cmake before the tools can understand the PoseLib code.

Reply:
Sorry for a bit misleading wording, i have meant that one could use a `Dependency Providers` mechanism to override default behavior and to point cmake into standalone PoseLib instance which state could be preserved/shared between multiple configurations. The part about "ignoring" was to highlight the most probable action for a user facing the scenario i proposed to consider.

Regarding your concern about IDE features that becomes available only after cmake project is configured - i agree that this is another angle of the same issue.

Reply:
I now converged to introducing a "FETCH_POSELIB" option, which allows to consume poselib using find_package.